src/app/modules/test-navigator/test-navigator.component.ts

```diff
@@ -49,6 +50,10 @@ export class TestNavigatorComponent implements OnInit, OnDestroy {
     }
   }
 
+  onFiltersChanged(state: FilterState) {
+    this.model.forEach((node) => (node as TestNavigatorTreeNode).setVisible(filterFor(state, node)));
```

Comment:
what about filters already active (like the general workspace filter, filtering e.g. java files)? is their visibility overwritten?

Reply:
if no filters are active as per the `FilterState` parameter, the default will still be that anything other than the known Test-Editor files, as well as directories, are filtered out. This test is meant to check for this behavior: https://github.com/test-editor/web-test-navigator/pull/12/files#diff-d700870cfbf0f30595a9b7615c9c395eR71